### PR TITLE
Fully implement partially implemented AccountMethods

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
# Description

This PR is part 1 for #405 and implements missing properties and entity parameters. No new methods will be added in this PR—that’ll be the task for a follow-up PR.

# Type of Change

- New feature
- Documentation

# Breaking Changes

- `AccountMethods#verifyCredentials` now returns `CredentialAccount` instead of `Account`
- `AccountMethods#updateCredentials` now returns `CredentialAccount` instead of `Account`
- `AccountMethods#updateCredentials` got a few new parameters. As none of them are optional, this will break existing calls, but they can easily be added.
- `AccountMethods#getStatuses` gained two now parameters `excludeReblogs` and `filterByTaggedWith`. Both are optional, like many existing parameters, but they are intertwined with other parameters. So if you use Java, or used Kotlin without named parameters here, your calls will likely or may return wrong results if you supply a Boolean to the wrong field.

# How Has This Been Tested?

New unit tests and updates for existing unit tests

# Mandatory Checklist

- [x] I ran `gradle check` and there were no errors reported
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [x] I have added KDoc documentation to all public methods

### CODE REVIEW

File: bigbone/src/main/kotlin/social/bigbone/api/method/AccountMethods.kt [hunk 1]
```diff
@@ -153,19 +250,37 @@ class AccountMethods(private val client: MastodonClient) {
...
      * @param accountId ID of the account to follow
+     * @param includeReblogs Receive this account’s reblogs in home timeline? Defaults to true if null.
+     * @param notifyOnStatus Receive notifications when this account posts a status? Defaults to false if null.
+     * @param filterForLanguages Filter received statuses for these languages. ISO 639-1 language two-letter codes.
+     * If not provided, you will receive this account’s posts in all languages.
+     *
      * @see <a href="https://docs.joinmastodon.org/methods/accounts/#follow">Mastodon API documentation: methods/accounts/#follow</a>
      */
-    fun followAccount(accountId: String): MastodonRequest<Relationship> {
+    fun followAccount(
```
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/api/method/AccountMethods.kt:270]: Here we should add `@JvmOverloads`.

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAccountMethods.kt
Comment (andregasser) [inline bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAccountMethods.kt:1]: This file should be checked for missing `@JvmOverloads` annotations.

File: bigbone/src/main/kotlin/social/bigbone/api/method/AccountMethods.kt
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/api/method/AccountMethods.kt:1]: This file should be checked for missing `@JvmOverloads` annotations.

File: bigbone/src/main/kotlin/social/bigbone/api/method/AccountMethods.kt (hunk 1 above)
Comment (PattaFeuFeu) [reply bigbone/src/main/kotlin/social/bigbone/api/method/AccountMethods.kt:271]: ✅

File: bigbone/src/main/kotlin/social/bigbone/api/method/AccountMethods.kt
Comment (PattaFeuFeu) [reply bigbone/src/main/kotlin/social/bigbone/api/method/AccountMethods.kt:1]: ✅

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAccountMethods.kt
Comment (PattaFeuFeu) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAccountMethods.kt:1]: ✅

**Review by andregasser — APPROVED** [review]
All good! Thank you :-)

**Pull request merged**